pkg/yqlib/decoder_toml.go

```diff
@@ -269,6 +269,11 @@ func (dec *tomlDecoder) processTable(currentNode *toml.Node) (bool, error) {
 
 	hasValue := dec.parser.NextExpression()
 	if !hasValue {
```

Comment:
It seems like this whole block could move _after_ the check below where the code already checks for an empty table, then the code will have already exited if it was an empty table.